src/Sql/Sql/ServiceObjective/Service/AzureSqlServerServiceObjectiveAdapter.cs

```diff
@@ -69,34 +82,69 @@ public AzureSqlServerServiceObjectiveModel GetServiceObjective(string resourceGr
         public List<AzureSqlServerServiceObjectiveModel> ListServiceObjectives(string resourceGroupName, string serverName)
         {
-            var resp = Communicator.List(resourceGroupName, serverName);
+            var server = ServerCommunicator.Get(resourceGroupName, serverName);
+            var capabilities = CapabilitiesCommunicator.Get(server.Location);
 
-            return resp.Select((s) =>
-            {
-                return CreateServiceObjectiveModelFromResponse(resourceGroupName, serverName, s);
-            }).ToList();
+            return (
+                from sv in FilterByName(capabilities.SupportedServerVersions, server.Version)
+                from e in sv.SupportedEditions
+                from slo in e.SupportedServiceLevelObjectives
+                select CreateServiceObjectiveModelFromResponse(e, slo, resourceGroupName, serverName)).ToList();
         }
 
         /// <summary>
-        /// Convert a Management.Sql.LegacySdk.Models.ServiceObjective to AzureSqlDatabaseServerServiceObjectiveModel
+        /// Convert a SLO capability to AzureSqlDatabaseServerServiceObjectiveModel
         /// </summary>
-        /// <param name="resourceGroupName">The resource group the server is in</param>
-        /// <param name="serverName">The name of the server</param>
-        /// <param name="resp">The management client ServiceObjective response to convert</param>
+        /// <param name="edition">The edition</param>
+        /// <param name="slo">The service objective</param>
+        /// <param name="resourceGroupName">The resource group name</param>
+        /// /// <param name="serverName">The server name</param>
         /// <returns>The converted ServiceObjective model</returns>
-        private static AzureSqlServerServiceObjectiveModel CreateServiceObjectiveModelFromResponse(string resourceGroupName, string serverName, Management.Sql.Models.ServiceObjective resp)
+        private static AzureSqlServerServiceObjectiveModel CreateServiceObjectiveModelFromResponse(
+            EditionCapability edition,
+            ServiceObjectiveCapability slo,
+            string resourceGroupName = null,
+            string serverName = null)
         {
-            AzureSqlServerServiceObjectiveModel slo = new AzureSqlServerServiceObjectiveModel()
+            return new AzureSqlServerServiceObjectiveModel()
             {
                 ResourceGroupName = resourceGroupName,
                 ServerName = serverName,
-                ServiceObjectiveName = resp.ServiceObjectiveName,
-                IsDefault = resp.IsDefault,
-                IsSystem = resp.IsSystem,
-                Description = resp.Description,
-                Enabled = resp.Enabled
+                ServiceObjectiveName = slo.Name,
+                IsDefault = slo.Status == CapabilityStatus.Default,
+                IsSystem = string.Equals(edition.Name, "System", StringComparison.OrdinalIgnoreCase),
```

Comment:
I plan to add System to DatabaseEditions (in SQL SDK) to eliminate this inline string